### PR TITLE
Update config

### PR DESCRIPTION
**Pull request by dbrockman**
- Test with node 4 and 6 on travis.
- Use official [`comma-dangle`](http://eslint.org/docs/rules/comma-dangle) rule instead of our own from the springworks plugin. The `comma-dangle` rule is auto fixable now.
- Activate [`no-whitespace-before-property`](http://eslint.org/docs/rules/no-whitespace-before-property) rule
- Activate [`no-unsafe-finally`](http://eslint.org/docs/rules/no-unsafe-finally) rule
- Activate [`no-useless-computed-key`](http://eslint.org/docs/rules/no-useless-computed-key) rule
- Activate [`no-useless-escape`](http://eslint.org/docs/rules/no-useless-escape) rule.
- Replace the `no-duplicates` rule from the import plugin with the built in [`no-duplicate-imports`](http://eslint.org/docs/rules/no-duplicate-imports) rule.
- Add tests that spawn `eslint` child processes. This will ensure that the config is correct. The tests will fail if a rule is misspelled or the config is invalid.
- Move linter config to js files. There is no need to have them split into a js and a json file anymore.
- Setup semantic-release

### CODE REVIEW

**Comment by dbrockman** [comment]
- [ ] configure semantic-release

File: .npmignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-examples
```
Comment (mferno) [inline .npmignore]: Does the examples stuff belong to this commit?

Comment (dbrockman) [inline .npmignore]: I replaced the examples with tests. The examples where not extensive enough to be useful. The test will start `eslint` in a new child process to make sure the config is valid.

Comment (mferno) [inline .npmignore]: Ok 👍

**Comment by dbrockman** [comment]
The new rules (linked in description) are all pretty common sense. Should not be controversial in any way.

**Pull request merged**